src/mail-app/mail/view/LabelsPopup.ts

```diff
@@ -116,7 +116,11 @@ export class LabelsPopup implements ModalComponent {
 
 	oncreate(vnode: VnodeDOM) {
 		this.dom = vnode.dom as HTMLElement
-		showDropdown(this.origin, this.dom, (this.labels.length + 1) * size.button_height + size.vpad_small * 2, this.width).then(() => {
+		const labelHeight = (this.labels.length + 1) * size.button_height + size.vpad_small * 2
+		//restrict label height to showing maximum 6 labels to avoid overflow
+		const labelMaxHeight = 7 * size.button_height + size.vpad_small * 2
+		const height = labelHeight > labelMaxHeight ? labelMaxHeight : labelHeight
+		showDropdown(this.origin, this.dom, height, this.width).then(() => {
```

Comment:
```suggestion
        // restrict label height to showing maximum 6 labels to avoid overflow
        const displayedLabels = Math.max(this.labels.length, 6)
        
		const height = (displayedLabels + 1) * size.button_height + size.vpad_small * 2
		showDropdown(this.origin, this.dom, height, this.width).then(() => {
```